tests/php/Command/Room/CreateTest.php

```diff
@@ -286,17 +290,18 @@ public function validProvider(): array {
 	 * @dataProvider invalidProvider
 	 */
 	public function testInvalid(array $input, string $expectedOutput): void {
-		$this->manager
-			->method('createGroupRoom')
-			->willReturnCallback(function (string $name = ''): Room {
-				return $this->roomMockContainer->create(['name' => $name, 'type' => Room::GROUP_CALL]);
-			});
 
-		$this->manager
-			->method('createPublicRoom')
-			->willReturnCallback(function (string $name = ''): Room {
-				return $this->roomMockContainer->create(['name' => $name, 'type' => Room::PUBLIC_CALL]);
-			});
+		if ($input['name'] !== 'PHPUnit Test Room') {
+			$this->roomService
+				->method('createConversation')
+				->willThrowException(new \InvalidArgumentException('name'));
```

Comment:
I feel like this is an hack (i.e. we would create another technical debt here). The fact that behaviour differs (for invalid names a exception is thrown, for anything else not; just like with `UpdateTest` vs. `CreateTest` - for the latter it's an exception, for `UpdateTest` it's not) is a problem.

Reply:
Well this is just temporarily until the other validations are moved from the controller to the service

Reply:
type and object data also throws now

Reply:
Very good :+1: 

Only one last thing: We shouldn't hard code the valid room name in the unit test. How about the following and adding a third data parameter to `invalidProvider` for those data sets which are expected to fail with an exception:

```php
	public function testInvalid(array $input, string $expectedOutput, ?Exception $expectedException = null): void {
		if ($expectedException !== null) {
			$this->roomService
				->method('createConversation')
				->willThrowException($expectedException);
		} else {
			$this->roomService
				->method('createConversation')
				->willReturnCallback(function (int $type, string $name): Room {
					return $this->roomMockContainer->create(['name' => $name, 'type' => $type]);
				});
		}
```

Reply:
well if okay for you I would skip that for now, until all parts have been moved over to RoomServce and then we can revisit the test structure.

Reply:
Sounds reasonable :+1: